prov/gni/include/gnix_cq.h

```diff
@@ -45,6 +45,9 @@
 #define GNIX_CQ_DEFAULT_FORMAT struct fi_cq_entry
 #define GNIX_CQ_DEFAULT_SIZE   256
 
+/* forward declaration */
+struct gnix_fid_ep;
```

Comment:
Additionally, usually the resolution for circular includes ends up resulting in some other header file being created that 'a' and 'b' include and removes the dependency of 'a' on 'b', or 'b' on 'a'